src/keys/bip39.rs

```diff
@@ -36,8 +36,10 @@ use bip39::{Language, Mnemonic, MnemonicType, Seed};
 
 use super::{any_network, DerivableKey, DescriptorKey, GeneratableKey, GeneratedKey, KeyError};
 
+/// Type for a BIP39 mnemonic with an optional passphrase
 pub type MnemonicWithPassphrase = (Mnemonic, Option<String>);
 
+#[cfg_attr(docsrs, doc(cfg(feature = "keys-bip39")))]
```

Comment:
Just a question to understand how this works, if `docsrs` attr is set then this line will generate docs for this  block with a message that it only applies to feature `keys-bip39` ? and without this line docs would only be generated if the `keys-bip39` feature were enabled?

Reply:
Not really, that's just to add the "badge" that says "this is only available if X feature is enabled".

The problem is that [this thing is still unstable](https://doc.rust-lang.org/beta/unstable-book/language-features/doc-cfg.html), so you can only build the project with the nightly toolchain if you want to use it. So instead we only apply it if the `docsrs` cfg is enabled, and we only do that on nightly.

You can see in `lib.rs` that we declare the unstable feature, but again we only do it whenever the `docsrs` cfg is present:

https://github.com/bitcoindevkit/bdk/blob/8d6f67c7640d1a64db18b72ff75afaea2b3daf91/src/lib.rs#L27-L29

And lastly we declare in our `Cargo.toml` that [docs.rs](docs.rs) should build the docs using that custom cfg argument. They always build using nightly, so that's not a problem:

https://github.com/bitcoindevkit/bdk/blob/8d6f67c7640d1a64db18b72ff75afaea2b3daf91/Cargo.toml#L84-L89

Reply:
Generally you can just add that to the line where you declare the module and it automatically applies it to every struct declared by that module:

https://github.com/bitcoindevkit/bdk/blob/8d6f67c7640d1a64db18b72ff75afaea2b3daf91/src/keys/mod.rs#L48-L49

But I noticed that it doesn't work for traits that are implemented inside the module, so the docs were looking a bit misleading because it seemed like for instance you could always generate `Mnemonic`s even without enabling the `keys-bip39` feature.. so I just added that attribute manually to every relevant piece of code

Reply:
ah ok thanks, now it all makes sense. So many things to learn with all the rust tools.